dnsmasq_interface.c

```diff
@@ -188,19 +191,6 @@ void FTL_forwarded(unsigned int flags, char *name, struct all_addr *addr, int id
 		// Check UUID of this query
 		if(queries[i].id == id)
 		{
-			// Detect if we cached the <CNAME> but need to ask the upstream
-			// servers for the actual IPs now
-			if(queries[i].status == QUERY_CACHE)
-			{
-				// Fix counters
-				counters.cached--;
-				validate_access("overTime", queries[i].timeidx, true, __LINE__, __FUNCTION__, __FILE__);
-				overTime[queries[i].timeidx].cached--;
-
-				// Mark this query again as (temporarily) unknown
-				counters.unknown++;
-				queries[i].complete = false;
-			}
```

Comment:
The logic of this was a bit strange. It is triggered in the following situation: Assume a domain `a.com` is a CNAME which is cached and has a very long TTL. It point to another domain `server.a.com` which has an A record but this has a much lower TTL.

If you now query `a.com` and then again after some time, you end up in a situation where `dnsmasq` can answer the first level of the DNS result (the CNAME) from cache, hence the status of this query is marked as "answered from cache" in `FTLDNS`. However, for `server.a.com` wit the much shorter TTL, we still have to forward something and ask the upstream server for the final IP address. This code section was meant to acknowledge this by reverting some of the already set query parameters and preparing it to be finishable by a part of the code (the one that marks queries as being forwarded.

I removed this because it is a strange exception and has to be done in a very unlucky place and is likely to get forgotten in the future when we introduce new functionality. Also, it has some problems with the reply timer which could, however, be resolved after some thinking. So, we could keep it, but then we should investigate if there is a better place to put this (I' not sure there is one).